### PR TITLE
Add assertion to make sure data and kernel are using the same SYCL context

### PR DESCRIPTION
**Pull request by reazulhoque**
This PR ensures USM data and SYCL kernels are using the same SYCL context.

### CODE REVIEW

File: numba_dppy/compiler.py [hunk 1]
```diff
@@ -493,6 +494,12 @@ def _unpack_argument(
 
         if isinstance(ty, types.Array):
             if hasattr(val.base, "__sycl_usm_array_interface__"):
+                assert (
+                    sycl_queue.get_sycl_context() == val.base._queue.get_sycl_context()
```
Comment (diptorupd) [inline numba_dppy/compiler.py:498]: Couple of issues:

a) The behavior is going to change when we move to the compute-follows-data design. In the current setup, I think the check is fine, but we can also do a copy through host if we want.
b) dpctl 0.8.0 introduced a new property `sycl_context` in the `MemoryUSMShared` class specifically for this scenario.

We can either merge the change and I can update it along with any other change for dpctl 0.8.0, or we hold off mering till the dpctl 0.8.0 chnages PR is ready.

Comment (reazulhoque) [reply numba_dppy/compiler.py:498]: I would like to propose that we merge it now. Once 0.8 is ready the API can be easily updated as APIs will need to be updated throughout Numba_dppy.

**Comment by PokhodenkoSA** [comment]
@diptorupd @reazulhoque 
Is it actual yet?

**Review by PokhodenkoSA — CHANGES_REQUESTED** [review]
Need to be updated.

**Comment by reazulhoque** [comment]
dppy.kernel has CFD implemented. We can get rid of these checks.

**Pull request closed without merging**